c/sedona-geos/src/st_buffer.rs

```diff
@@ -115,6 +151,115 @@ fn invoke_scalar(
+        ColumnarValue::Scalar(_) => Ok(None),
+        _ => Err(DataFusionError::Execution(format!(
+            "Expected scalar bufferStyleParameters, got: {:?}",
+            arg
+        ))),
+    }
+}
+
+fn parse_buffer_params(params_str: Option<&str>) -> Result<BufferParams> {
+    let Some(params_str) = params_str else {
+        return BufferParams::builder()
+            .build()
+            .map_err(|e| DataFusionError::External(Box::new(e)));
+    };
+
+    let mut params_builder = BufferParams::builder();
+
+    for param in params_str.split_whitespace() {
+        let Some((key, value)) = param.split_once('=') else {
+            return Err(DataFusionError::Execution(format!(
+                "Missing value for buffer parameter: {}",
+                param
+            )));
+        };
+
+        match key.to_lowercase().as_str() {
+            "endcap" => {
+                params_builder = params_builder.end_cap_style(parse_cap_style(value)?);
+            }
+            "join" => {
+                params_builder = params_builder.join_style(parse_join_style(value)?);
+            }
+            "side" => {
+                params_builder = params_builder.single_sided(parse_side(value)?);
+            }
+            "mitre_limit" | "miter_limit" => {
+                let limit: f64 = parse_number(value, "mitre_limit")?;
+                params_builder = params_builder.mitre_limit(limit);
+            }
+            "quad_segs" | "quadrant_segments" => {
+                let segs = parse_number(value, "quadrant_segments")?;
+                params_builder = params_builder.quadrant_segments(segs);
+            }
+            _ => {
+                return Err(DataFusionError::Execution(format!(
+                    "Invalid buffer parameter: {} (accept: 'endcap', 'join', 'mitre_limit', 'miter_limit', 'quad_segs' and 'side')",
+                    key.to_lowercase()
+                )));
+            }
+        }
+    }
+
+    params_builder
+        .build()
+        .map_err(|e| DataFusionError::External(Box::new(e)))
+}
+
+fn parse_cap_style(value: &str) -> Result<CapStyle> {
+    match value.to_lowercase().as_str() {
+        "round" => Ok(CapStyle::Round),
+        "flat" | "butt" => Ok(CapStyle::Flat),
+        "square" => Ok(CapStyle::Square),
+        _ => Err(DataFusionError::Execution(format!(
+            "Invalid endcap style: '{}'. Valid options: round, flat, butt, square",
+            value
+        ))),
+    }
+}
+
+fn parse_join_style(value: &str) -> Result<JoinStyle> {
+    match value.to_lowercase().as_str() {
+        "round" => Ok(JoinStyle::Round),
+        "mitre" | "miter" => Ok(JoinStyle::Mitre),
+        "bevel" => Ok(JoinStyle::Bevel),
+        _ => Err(DataFusionError::Execution(format!(
+            "Invalid join style: '{}'. Valid options: round, mitre, miter, bevel",
+            value
+        ))),
+    }
+}
+
+fn parse_side(value: &str) -> Result<bool> {
```

Comment:
```suggestion
fn is_single_sided(value: &str) -> Result<bool> {
```
nit: I think `is_single_sided` would be a better name here.
`parse_side` would be more appropriate if it was returning a `Side` enum (with variants BOTH, LEFT and RIGHT) and this enum could have a method `is_single_sided`